src/plugin/memory-webview-main.ts

```diff
@@ -301,4 +312,26 @@ export class MemoryWebview implements vscode.CustomReadonlyEditorProvider {
 
         this.setMemoryViewSettings(ctx.messageParticipant, { visibleColumns });
     }
+
+    protected async storeMemory(storeArguments: StoreMemoryArguments): Promise<void> {
+        // Even if we disable the command in VS Code through enablement or when condition, programmatic execution is still possible.
+        // However, we want to fail early in case the user tries to execute a disabled command
+        if (!this.memoryProvider.createContext().canRead) {
+            throw new Error('Cannot read memory, no valid debug session.');
+        }
+        return vscode.commands.executeCommand(StoreCommandType, storeArguments);
+    }
+
+    protected async applyMemory(): Promise<MemoryOptions> {
+        // Even if we disable the command in VS Code through enablement or when condition, programmatic execution is still possible.
+        // However, we want to fail early in case the user tries to execute a disabled command
+        if (!this.memoryProvider.createContext().canWrite) {
+            throw new Error('Cannot write memory, no valid debug session.');
+        }
+        return vscode.commands.executeCommand(ApplyCommandType);
+    }
```

Comment:
For a follow-up: since we're sending a context to the view that includes information about the session that that window is associated with, these requests could handle that association and check the relevant session, rather than defaulting to the active session. Re: #97.

Reply:
This is a very good point and I believe that should be handled as part of #97 as we definitely want to do this for all requests (read, write, store, apply, get variables).